components/permissions/contexts/brave_wallet_permission_context.cc

```diff
@@ -211,9 +215,11 @@ void BraveWalletPermissionContext::GetAllowedAccounts(
   }
 
   auto* web_contents = content::WebContents::FromRenderFrameHost(rfh);
-  // Fail the request came from 3p origin.
-  if (web_contents->GetMainFrame()->GetLastCommittedOrigin() !=
-      rfh->GetLastCommittedOrigin()) {
+  if (rfh->GetParent() &&
```

Comment:
`// Fail the request came from a different eTLD+1 origin.` is missing for this function

Reply:
Fixed in 9d0ff1f736f79bae8aed8eaf90607b24469e8d6b